### PR TITLE
Added MergeSort.java

### PR DESCRIPTION
**Pull request by the-cybersapien**
Added java implementation for MergeSort.

### CODE REVIEW

File: Sorting Algorithms/Merge Sort Algorithm/Java/MergeSort.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import java.util.Scanner;
+
+/**
+ * Theory of Programming
+ * Created December 25, 2016 by cybersapien
```
Comment (VamsiSangam) [inline Sorting Algorithms/Merge Sort Algorithm/Java/MergeSort.java]: Please add "http://theoryofprogramming.com/2015/01/29/merge-sort-algorithm/" here and for the sake of consistency, style your comments in this fashion -

```
/* ===== ===== =====

Theory of Programming

<mention_the_algorithm_or_data_structure_name>
<give_the_link_to_corresponding_theory_of_programming_post>
GitHub - https://github.com/VamsiSangam/theoryofprogramming
Code Contributor - <contributor_name>, <other_contributor_name_if_any>

===== ===== ===== */
```

File: Sorting Algorithms/Merge Sort Algorithm/Java/MergeSort.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        mergesort(array,0, size-1);
+        //Print the final answer to the User:
+        System.out.println("Sorted Array: ");
+        for (int x: array){
+            System.out.print(x + " ");
+        }
+        System.out.println();
+    }
+
+    private static void mergesort(int array[], int low, int high){
```
Comment (VamsiSangam) [inline Sorting Algorithms/Merge Sort Algorithm/Java/MergeSort.java]: Requesting anther minor change. Please name your method ```mergeSort```.

File: Sorting Algorithms/Merge Sort Algorithm/Java/MergeSort.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            if (aux[i]<=aux[j]){
+                array[k] = aux[i];
+                i++;
+            } else{
+                array[k] = aux[j];
+                j++;
+            }
+            k++;
+        }
+        while (i<=mid){
```
Comment (VamsiSangam) [inline Sorting Algorithms/Merge Sort Algorithm/Java/MergeSort.java]: Please add spaces on either ends of operators. It makes the code much readable.

**Comment by the-cybersapien** [comment]
Fixed the code to match your style. Please Review

File: Sorting Algorithms/Merge Sort Algorithm/Java/MergeSort.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        }
+        System.out.println();
+    }
+
+    private static void mergeSort(int array[], int low, int high){
+        //Check if low is smaller than higher
+        // this is the exit condition for the recursive solution
+        if (low < high){
+            //get the index of the element in the middle
+            int mid = low + (high-low)/2;
```
Comment (VamsiSangam) [inline Sorting Algorithms/Merge Sort Algorithm/Java/MergeSort.java]: Requesting another minor change. I requested for spaces on either side of every binary operator in the whole program not just for the lines mentioned in the previous review.

**Comment by the-cybersapien** [comment]
Sorry, I thought I had taken care of all the binary operators. Missed a couple lines. 
Please Review.

**Comment by VamsiSangam** [comment]
Sorry if I am being too particular. Your code will be displayed on a tutorial website, so I have to be very particular about readability. Will review your code now. And yes, add your name to [CONTRIBUTORS.md](https://github.com/VamsiSangam/theoryofprogramming/blob/master/CONTRIBUTORS.md) file. 😄

**Comment by the-cybersapien** [comment]
Added my name to Contributors.md too.

**Review by VamsiSangam — COMMENTED** [review]
Please add new lines before and after loops Example -

```

for loop {

}

```

and if there's a comment just above the loop, add a new line before the comment -

```

// comment
for loop {

}

```

**Review by VamsiSangam — APPROVED** [review]

**Comment by the-cybersapien** [comment]
Added newLines before and After loops. :+1: 
I follow the Android Code Style so There's a lot of differences :laughing:

**Comment by VamsiSangam** [comment]
Hey you write pretty decent code man! Thanks for bearing me with all those minor changes. And thanks a lot for your quick contribution! Will merge your PR now. You should be able to see your contribution live in the [merge sort](http://theoryofprogramming.com/2015/01/29/merge-sort-algorithm/) post in 5-10 mins.

**Pull request merged**